core/src/test/java/io/github/wysohn/triggerreactor/core/script/interpreter/TestInterpreter.java

```diff
@@ -1980,6 +1980,85 @@ public Integer execute(Timings.Timing timing, Map<String, Object> vars, Object c
         interpreter.startWithContext(null);
     }
 
+    @Test
+    public void testImportAs() throws Exception {
+        Charset charset = StandardCharsets.UTF_8;
+        String text = "IMPORT io.github.wysohn.triggerreactor.core.script.interpreter.TestInterpreter$TheTest as testClass;"
+            + "IMPORT io.github.wysohn.triggerreactor.core.script.interpreter.TestInterpreter$TestEnum as testEnum;"
+            + "#TEST testClass;"
+            + "#TEST2 testClass.staticTest();"
+            + "#TEST3 testClass().localTest();"
+            + "#TEST4 testClass.staticField;"
+            + "#TEST5 testEnum.IMTEST;";
+
+        Lexer lexer = new Lexer(text, charset);
+        Parser parser = new Parser(lexer);
+
+        Node root = parser.parse();
+        Map<String, Executor> executorMap = new HashMap<>();
+        executorMap.put("TEST", new Executor() {
+
+            @Override
+            public Integer execute(Timings.Timing timing, Map<String, Object> vars, Object context,
+                                   Object... args) throws Exception {
+
+                assertEquals(TheTest.class, args[0]);
+                return null;
+            }
+
+        });
+        executorMap.put("TEST2", new Executor() {
+
+            @Override
+            public Integer execute(Timings.Timing timing, Map<String, Object> vars, Object context,
+                                   Object... args) throws Exception {
+
+                assertEquals("static", args[0]);
+                return null;
+            }
+
+        });
+        executorMap.put("TEST3", new Executor() {
+
+            @Override
+            public Integer execute(Timings.Timing timing, Map<String, Object> vars, Object context,
+                                   Object... args) throws Exception {
+
+                assertEquals("local", args[0]);
+                return null;
+            }
+
+        });
+        executorMap.put("TEST4", new Executor() {
+
+            @Override
+            public Integer execute(Timings.Timing timing, Map<String, Object> vars, Object context,
+                                   Object... args) throws Exception {
+
+                assertEquals("staticField", args[0]);
+                return null;
+            }
+
+        });
+        executorMap.put("TEST5", new Executor() {
+
+            @Override
+            public Integer execute(Timings.Timing timing, Map<String, Object> vars, Object context,
+                                   Object... args) throws Exception {
+
+                assertEquals(TestEnum.IMTEST, args[0]);
+                return null;
+            }
+
+        });
+
+        Interpreter interpreter = new Interpreter(root);
+        interpreter.setExecutorMap(executorMap);
+        interpreter.setTaskSupervisor(mockTask);
+
+        interpreter.startWithContext(null);
```

Comment:
This is good, but can we use the mock Executor instance instead?

```
Executor mockExecutor = mock(Executor.class);
executorMap.put("TEST5", mockExecutor);

...
interpreter.start...
...

verify(mockExecutor).execute(any(), anyMap(), any(), eq(new Object[]{TestEnum.IMTEST}));
```

Reply:
Do we have to create much mock executors I guess? There are five executors to test.

Or we can put those into the variables and check the values match.

Reply:
For code like:
```java
String text = "IMPORT io.github.wysohn.triggerreactor.core.script.interpreter.TestInterpreter$TheTest as testClass;"
  + "IMPORT io.github.wysohn.triggerreactor.core.script.interpreter.TestInterpreter$TestEnum as testEnum;"
  + "clazz = testClass;"
  + "staticMethod = testClass.staticTest();";

// ...
Map<String, Object> vars = interpreter.getVars();

assertEquals(TheTest.class, vars.get("clazz"));
assertEquals(TheTest.staticTest(), vars.get("staticMethod"));
```

Reply:
> Do we have to create much mock executors I guess? There are five executors to test.
> 
> Or we can put those into the variables and check the values match.

It's really up to you.

The downside of using the lambda function and putting the assertion inside the method body is that it might be the case that the given Executor (the lambda) never gets called, so the assertion is also not checked either.

verify() make sure that the method invocation on the mock instance is called once (or you can even check 0 times or more than 2 times) and also the value of the parameters.

any() is like you don't care what the value would be, and eq() is what you expect for the arguments of the method invocation to receive.

Reply:
> For code like:
> 
> ```java
> String text = "IMPORT io.github.wysohn.triggerreactor.core.script.interpreter.TestInterpreter$TheTest as testClass;"
>   + "clazz = testClass;"
>   + "staticMethod = testClass.staticTest();";
> 
> // ...
> Map<String, Object> vars = interpreter.getVars();
> 
> assertEquals(TheTest.class, vars.get("clazz"));
> assertEquals(TheTest.staticTest(), vars.get("staticMethod"));
> ```

This is completely valid. In fact, your previous test also works, but it's just the case that it didn't cover the case where the Executor is never invoked.

Reply:
I think it's fully care the test now (I pushed new commit with the second one.).

To be honest, Mockito did not take care about ordered list when I mocked executor up. I also used [InOrder](https://github.com/mockito/mockito/blob/master/src/main/java/org/mockito/InOrder.java) verification but Test didn't invoke except for the first case.

Reply:
Yeah Mockito sometimes works strangely.

Sometimes declaring the variable first and use that as expected value fixed the problem for List, but not sure if that would work for the array.

Reply:
oh wait you are talking about something else.

If the mockito doesn't catch the invocation, it's probably because it really wasn't invoked. But the error message can be vague.

A recommendation would be to separate the tests into their own method so it's easier to see where the test failed.

Reply:
Thanks for the suggestion, but I fixed it :D

I forgot to get the executor to return null.